backend/apps/core/cache/invalidation.py

```
@@ -0,0 +1,119 @@
+    Deletes all cache keys associated with the given tag (or pattern).
+    """
+    if "*" in tag:
+        matching_tags = []
+        try:
+            from django_redis import get_redis_connection
+            con = get_redis_connection("default")
+            pattern = f"cache_tag:{tag}"
+            for k in con.scan_iter(match=pattern):
+                k_str = k.decode("utf-8") if isinstance(k, bytes) else k
+                matching_tags.append(k_str.replace("cache_tag:", ""))
+        except Exception:
```

Ruff (0.15.21)

[warning] Do not catch blind exception: `Exception`

(BLE001)